File: LastSemesterInfo/AYDA Laboratorio/AYDA_Examen_1_student_distribution_(A2026)/windows/longest_increasing_subsequence/python/template.py
import sys

def longestIncreasingSubsequence(arr):
    return 0

if __name__ == '__main__':
    n = int(sys.stdin.readline().strip())
    arr = []
    for _ in range(n):
        arr.append(int(sys.stdin.readline().strip()))

    result = longestIncreasingSubsequence(arr)

    with open('solution.txt', 'w') as f:
        f.write(str(result) + '\n')

File: LastSemesterInfo/AYDA Laboratorio/AYDA_Examen_1_student_distribution_(A2026)/windows/longest_increasing_subsequence/rust/template.rs
use std::io::{self, BufRead};
use std::fs::File;
use std::io::Write;

fn longest_increasing_subsequence(arr: &[i32]) -> i32 {
    0
}

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();

    let n: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    let mut arr: Vec<i32> = Vec::with_capacity(n);

    for _ in 0..n {
        let item: i32 = lines.next().unwrap().unwrap().trim().parse().unwrap();
        arr.push(item);
    }

    let result = longest_increasing_subsequence(&arr);

    let mut file = File::create("solution.txt").expect("The file could not be created");
    writeln!(file, "{}", result).expect("The file could not be written to");
}